spanner/spannertest/inmem.go

```diff
@@ -485,9 +485,21 @@ func (s *server) readTx(ctx context.Context, session string, tsel *spannerpb.Tra
 }
 
 func (s *server) ExecuteSql(ctx context.Context, req *spannerpb.ExecuteSqlRequest) (*spannerpb.ResultSet, error) {
-	// Assume this is probably a DML statement. Queries tend to use ExecuteStreamingSql.
+	// Assume this is probably a DML statement or a ping from the session pool.
+	// Queries normally use ExecuteStreamingSql.
 	// TODO: Expand this to support more things.
 
+	// If it is a single-use transaction we assume it is a query.
+	if req.Transaction == nil ||
```

Comment:
the proto getters work fine with nil receivers by design, so you can drop the first and third part of the clause.

```go
if req.Transaction.GetSelector() == nil || req.Transaction.GetSingleUse().GetReadOnly() != nil {
```

Reply:
Done.